docs/src/capi/explanation/in-place-upgrades.md

```diff
@@ -0,0 +1,160 @@
+# In-Place Upgrades
+
+Regularly upgrading the Kubernetes version of the machines in a cluster 
+is important. While rolling upgrades a popular strategy, certain situations 
+will require in-place upgrades:
+
+- Resource constraints (i.e. cost of additional machines).
+- Expensive manual setup process for nodes.
+
+## Annotations
+
+CAPI machines are considered immutable. Consequently, machines are replaced 
+instead of reconfigured.
+While CAPI doesn't support in-place upgrades, {{product}} CAPI does 
+by leveraging annotations for the implementation.
+For more information about CAPI design decisions, have a look at the 
+following:
+
+- [Machine immutability in CAPI][1]
+- [Kubernetes objects: `labels`][2]
+- [Kubernetes objects: `spec` and `status`][3]
+
+## Controllers
+
+In {{product}} CAPI, there are two main types of controllers that handle the 
+process of performing in-place upgrades:
+
+- Single Machine In-Place Upgrade Controller
+- Orchestrated In-Place Upgrade Controller
```

Comment:
```suggestion
- Single Machine Upgrade Controller
- Orchestrated Upgrade Controller
```
Can we simplify the names? We refer to them later as such in the next paragraph. 
Double check that this name is the same as the one in your pictures.

Reply:
I think it's better to leave them here, the pictures already have this in their names.

Reply:
Okay that's fine.